### ISSUE TITLE
`estimateGas` error message in Hardhat node

### ISSUE THREAD

**Issue by fvictorio** [nomiclabs/hardhat#1731]
A user got this output in their node:

![image](https://user-images.githubusercontent.com/417134/127056391-354affce-dedb-4eae-bb77-d77de11e821c.png)

I don't have reproduction steps, although it can probably be simulated locally by hardcoding some `throw` somewhere. The point is that a message like that is very confusing and we should either not show anything at all or show something better.

My guess of how to reproduce this:

1. Check the part of the code that throws when an unrecognized error happens
2. Hardcode that to make it always throw
3. Start an `hh node` and send an `eth_estimateGas`

**Comment by degoev** [comment]
have this issue too

**Comment by penguin4238** [comment]
have this issue too with apple m1  and node v18.4.0, hardhat 2.10.2

**Comment by gapon2401** [comment]
I have the same issue. Hardhat 2.11.0
I'm trying to test smartcontract locally.

My steps to reproduce:

1) Fork the mainnet:

A part of hardhat.config.ts
```
networks: {
    hardhat: {
      allowUnlimitedContractSize: true,
      forking: {
        url: "https://eth-mainnet.g.alchemy.com/v2/<ALCHEMY_KEY>",
      }
    },
    localhost: {
      allowUnlimitedContractSize: true,
      chainId: 31337,
      accounts: [`PRIVATE_KEY1`, `PRIVATE_KEY2`],
    },
}
```

2) Auth in metamask, connect to network localhost:8545 with chainId 31337
3) Add hardhat account in metamask with PRIVATE_KEY1
4) Deploy smartcontract
5) Trying to send eth to smartcontract in metamask. Button, which starts the transaction, is inactive

Here, what I have in console:

![SharedScreenshot](https://user-images.githubusercontent.com/4976144/188669019-59c7d1c5-4e16-486b-8372-c2d3a35c8af4.jpg)

**Comment by victorshevtsov** [comment]
I confirm. I'm facing the same issue.
Absolutely reproducible.

**Comment by victorshevtsov** [comment]
It looks like the issue is in MetaMask browser extension.

It works just fine with MetaMask for Android.
All the transactions to `hardhat node` goes well when performed from mobile wallet but not from a browser extension.

I tried Chrome and FireFox extensions.

**Comment by victorshevtsov** [comment]
Finally I got it!
It took almost whole day to figure out what is the cause.

The default `chainId` of Hardhat node is `31337` and this causes the issue.
I changed it to `1337` and everything works now.

```javascript
/// hardhat.config.js:

/** @type import('hardhat/config').HardhatUserConfig */
module.exports = {
  solidity: "0.8.17",
  networks: {
    hardhat: {
      chainId: 1337, // default is 31337
    }
  }
};
```

I'm sure the root of the problem is out there.
I hope somebody will figure it out and get it fixed eventually.

**Comment by nfts2me** [comment]
> Finally I got it! It took almost whole day to figure out what is the cause.
> 
> The default `chainId` of Hardhat node is `31337` and this causes the issue. I changed it to `1337` and everything works now.
> 
> ```js
> /// hardhat.config.js:
> 
> /** @type import('hardhat/config').HardhatUserConfig */
> module.exports = {
>   solidity: "0.8.17",
>   networks: {
>     hardhat: {
>       chainId: 1337, // default is 31337
>     }
>   }
> };
> ```
> 
> I'm sure the root of the problem is out there. I hope somebody will figure it out and get it fixed eventually.

As described here: 
https://github.com/NomicFoundation/hardhat/issues/3089#issuecomment-1317460141
I'm getting this error as well.
But changing the chainId didn't fix the issue.

**Comment by julianwells3510** [comment]
Yes, there’s some kind of givеаway going on. I got some Ethereum there yesterday for different wallets. It’s unlikely they’ll take it back from me, but it’s a pity I didn’t have more wallets and couldn’t get more. But maybe it’s for the best.

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/912a3c83-0225-4fd5-a01d-14a0ab2916b1" />

<img width="373" alt="Image" src="https://github.com/user-attachments/assets/627f6157-5e74-4711-8891-4ea87284d103" />
